pandas/core/arrays/base.py

```diff
@@ -530,8 +540,9 @@ def astype(self, dtype, copy=True):
 
         Returns
         -------
-        array : ndarray
-            NumPy ndarray with 'dtype' for its dtype.
+        array : ArrayLike
+            An ExtensionArray if dtype StringDtype or same as that of underlying array.
```

Comment:
is StringDtype a typo?

Reply:
oh i see, this is bc of the special-casing.  should be fixed by #41652

Reply:
clarified in next commit by saying "if dtype is StringDtype"